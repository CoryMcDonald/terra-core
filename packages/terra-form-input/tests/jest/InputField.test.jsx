import React from 'react';
import InputField from '../../src/InputField';

it('should render a default InputField component', () => {
  const textarea = <InputField label="Label" />;
  const wrapper = shallow(textarea);
  expect(wrapper).toMatchSnapshot();
});

it('should render a InputField with props', () => {
  const textarea = (
    <InputField
      defaultValue="Value"
      error="Text"
      help="Help"
      hideRequired
      inputAttrs={{
        name: 'test',
        type: 'number',
      }}
      inputId="test"
      isInline
      isInvalid
      isLabelHidden
      label="Label Test"
      labelAttrs={{
        className: 'label',
      }}
      onChange={() => {}}
      hideRequiredshowOptional
      value="Value"
    />
  );

  const wrapper = shallow(textarea);
  expect(wrapper).toMatchSnapshot();
});
