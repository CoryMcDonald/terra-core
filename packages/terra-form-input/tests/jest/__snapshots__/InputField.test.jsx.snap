// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a InputField with props 1`] = `
<Field
  error="Text"
  errorIcon={
    <IconError
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  help="Help"
  hideRequired={true}
  hideRequiredshowOptional={true}
  htmlFor="test"
  isInline={true}
  isInvalid={true}
  isLabelHidden={true}
  label="Label Test"
  labelAttrs={
    Object {
      "className": "label",
    }
  }
  required={false}
  showOptional={false}
>
  <Input
    defaultValue="Value"
    disabled={false}
    id="test"
    isInvalid={false}
    name="test"
    onChange={[Function]}
    required={false}
    type="number"
    value="Value"
  />
</Field>
`;

exports[`should render a default InputField component 1`] = `
<Field
  error={null}
  errorIcon={
    <IconError
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  help={null}
  hideRequired={false}
  isInline={false}
  isInvalid={false}
  isLabelHidden={false}
  label="Label"
  labelAttrs={Object {}}
  required={false}
  showOptional={false}
>
  <Input
    disabled={false}
    isInvalid={false}
    name={null}
    required={false}
  />
</Field>
`;
